libsolidity/formal/SMTChecker.cpp

```diff
@@ -768,6 +781,17 @@ void SMTChecker::checkCondition(
 			expressionsToEvaluate.emplace_back(var.second->currentValue());
 			expressionNames.push_back(var.first);
 		}
+		// Names for UFs are created later.
```

Comment:
This is getting rather complicated and should probably be split into multiple functions.